packages/loot-core/src/types/models/schedule.d.ts

```diff
@@ -25,6 +25,9 @@ export interface ScheduleEntity {
     }[];
     skipWeekend: boolean;
     start: string;
+    endMode: 'never' | 'after_n_occurrences' | 'on_date';
+    endOccurrences: number;
```

Comment:
Would be be possible to only store `endDate` regardless of `endMode`?

So when the user selects '5 occurrences' we work out the _date_ of the 5th occurrence and store that? Just thinking about saving a column but maybe its not worth the effort.

Reply:
I think it makes showing the future UI a lot easier. For example, if you add a schedule and then go edit the rule later, having a dedicated `endOccurrences` field means we can just use that for hydrating the UI. If we don't have that field, we have to have logic like "generate dates until we pass the end date" which could get pretty inefficient if someone inputs a large # of occurrences. Let me know if this makes sense!

Reply:
I'm not too fussed either way, Just a suggestion! But I wouldn't necessarily have thought it would hurt perf too badly even for v large dates, particularly as you only see one at a time.

Reply:
Yeah, thanks for the suggestion! I think I should have been more clear: I'm talking about the rules page which displays the same selector:
<img width="585" alt="image" src="https://github.com/actualbudget/actual/assets/9922514/256d66e3-cb5d-4c51-8dd8-eba46b84886a">
If we don't store `endOccurrences`, I think we'd have to recalculate it on this page by manually iterating until we hit `endDate`, no?

Reply:
Yeh you're correct; you'd probably have to iterate through the schedule executions till you reach 'end date'. Even then I dont think it would be too slow.

but given what you've got now works I wouldn't worry too much about it!